test/sqllogictest/types.slt

```diff
@@ -596,9 +596,79 @@ SELECT '1'::materialize.pg_catalog.int4
 1
 
 # tables are not types yet
-query error "pg_catalog.pg_enum" does not refer to a type
+query error type "pg_catalog.pg_enum" does not exist
 SELECT '1'::pg_catalog.pg_enum
 
+# relations can have the same name as built-in types
+statement ok
+CREATE VIEW int4 AS VALUES (1)
+
+query I
+SELECT * FROM int4
+----
+1
+
+# but within the same schema, types cannot have the same name as a relation
+statement error view "materialize.public.int4" already exists
+CREATE TYPE int4 AS (a int)
+
+# creating relations with the same name as an existing type is not allowed...
```

Comment:
This is a little more complex than this statement sounds because each relation has a type that is automatically created for it in PG––filed https://github.com/MaterializeInc/materialize/issues/23789 to track this